### PR TITLE
Follow-up the PR #4335 (Debug char, fix compile errors for lcd pins)

### PR DESCRIPTION
**Pull request by ghost**
Follow-up the PR #4335 (Debug char, fix compile errors for lcd pins)

It solves a [flickering when SD card is inserted or removed at using a ASCII LCD + `language_kana.h`](https://github.com/MarlinFirmware/Marlin/pull/4185#issuecomment-233195378).
This fix is by [@thinkyhead](https://github.com/MarlinFirmware/Marlin/pull/4335#issuecomment-233220856) and [@Blue-Marlin](This need the same correction for MAPPER_NON as lcd_strlen%28%29) and @AnHardt, thus I much thankful to you again.

...Or, does alternative solution for fix available? Because

> [Probably not the best or real solution:](https://github.com/MarlinFirmware/Marlin/pull/4335#issuecomment-233220856)

.
If it's true, please close this PR.

### CODE REVIEW

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -2741,7 +2741,15 @@ void lcd_update() {
...
-    if ((s[i] & 0xC0u) != 0x80u) j++;
+    #if ENABLED(MAPPER_NON)
+      #if ENABLED(DISPLAY_CHARSET_ISO10646_KANA)
+        if (s[i] != 0xDEu) j++;
+      #else
+        j++;
+      #endif
+    #else
+      if ((s[i] & 0xC0u) != 0x80u) j++;
+    #endif
```
Comment (AnHardt) [inline Marlin/ultralcd.cpp]: DE is a screen symbol at its own here.
We are pointing directly into the map of symbols.
No combining of the accent and the according character can be made.

```
-    if ((s[i] & 0xC0u) != 0x80u) j++;
 +    #if ENABLED(MAPPER_NON)
 +        j++;
 +    #else
 +      if ((s[i] & 0xC0u) != 0x80u) j++;
 +    #endif
```

should be more correct.

**Comment by ghost** [comment]
Thanks for the advice, I've updated.
And I've confirmed just now that it hasn't side effect in case of 
ASCII LCD + `language_kana.h`,
ASCII LCD + `language_en.h`,
ASCII LCD + `language_de.h`,
GLCD + `language_kana_utf8.h`, 
GLCD + `language_en.h`, 
GLCD + `language_de.h`.

**Comment by thinkyhead** [comment]
This is an improvement, because we do need to consider `MAPPER_NON`. However, the character `0xDE` (222) should not be counted as a printing character. For graphical displays we also might need to consider `MAPPER_ONE_TO_ONE`. I'm not sure, because that flag is used elsewhere.
```cpp
#if DISABLED(SIMULATE_ROMFONT) && ENABLED(DOGLCD)
  #if ENABLED(DISPLAY_CHARSET_ISO10646_1)
    #define MAPPER_ONE_TO_ONE
  #elif ENABLED(DISPLAY_CHARSET_ISO10646_5)
    #define MAPPER_ONE_TO_ONE
  #elif ENABLED(DISPLAY_CHARSET_ISO10646_KANA)
    #define MAPPER_ONE_TO_ONE
  #elif ENABLED(DISPLAY_CHARSET_ISO10646_GREEK)
    #define MAPPER_ONE_TO_ONE
  #endif
#else // SIMULATE_ROMFONT
  . . .
#endif
```

**Comment by thinkyhead** [comment]
Ah, I see the "non-printing 0xDE" character I was concerned about is shown as a printed character, so the code can be just as you have it here. Ok, then, good on us!

**Pull request merged**